ocpmodels/trainers/base_trainer.py

```diff
@@ -576,3 +685,39 @@ def _backward(self, loss):
             self.scaler.update()
         else:
             self.optimizer.step()
+
+    def save_results(self, predictions, results_file, keys):
+        if results_file is None:
+            return
+
+        results_file_path = os.path.join(
+            self.config["cmd"]["results_dir"],
```

Comment:
This directory probably won't exist unless all processes started at the exact same time, right? We create `results_dir` only for the master process.

Reply:
Master device creates the directory, but it's accessible by all devices. This gets created when the trainer is instantiated anyways, so it'll exist by the time this is called. By processes I'm assuming you mean GPUs, this is relevant for our DDP implementation.

Reply:
I was talking about the edge case where `results_dir` in master is say `results/2020-10-26-10-10-10`, but on another GPU is `results/2020-10-26-10-10-11`.. Just the master `results_dir` would get created and exist, right?

Reply:
Ah good point. Yeah only master gets created/exists. Not clear/obvious to me if this edge case occurs with how DDP is implemented, @anuroopsriram thoughts? Either way we should fix in case.

Reply:
Should we remove timestamps from the directory names?